addons/account/models/account_move.py

```diff
@@ -1991,6 +1996,24 @@ def _compute_tax_country_code(self):
         for record in self:
             record.tax_country_code = record.tax_country_id.code
 
+    @api.depends('company_id', 'partner_id', 'amount_total')
+    def _compute_partner_credit_warning(self):
+        for move in self:
+            move.with_company(move.company_id)
+            move_amount_company_currency = move.company_id.currency_id._convert(
+                move.amount_total, move.currency_id, move.company_id, move.date)
+            # To avoid overlaps, do not add the current invoice to the partner's credit
+            # if it is created from a sales order.
+            move.partner_updated_credit = move.partner_credit
+            original_order = self.env['sale.order'].sudo().search(
+                [('partner_id', '=', move.partner_id.id), ('invoice_ids', '=', move.id)])
+            if not original_order:
+                move.partner_updated_credit += move_amount_company_currency
+            move.show_partner_credit_warning = move.state == 'draft' and \
+                                               move.move_type == 'out_invoice' and \
+                                               move.company_id.account_credit_limit and \
+                                               move.partner_updated_credit > move.partner_credit_limit >= 0
```

Comment:
Do we need a field to compute this? Can't it be done as an `invisible` attribute?

Reply:
Since I have to compute the updated value, it was easier to compute `show_partner_credit_warning` here as well, rather than having a especially long `invisible` field in the view.
And it was also a suggestion I got here :
https://github.com/odoo/odoo/pull/83205#discussion_r823796665